libres/lib/res_util/block_fs.cpp

```diff
@@ -733,16 +568,13 @@ void block_fs_fwrite_file(block_fs_type *block_fs, const char *filename,
         throw std::runtime_error("tried to write to read only filesystem");
     std::lock_guard guard{block_fs->mutex};
 
-    file_node_type *file_node;
-    bool new_node = true;
     size_t min_size = data_size + file_node_header_size(filename);
 
-    file_node = block_fs_get_new_node(block_fs, filename, min_size);
+    BlockMeta block = block_fs_get_new_node(block_fs, filename, min_size);
 
     /* The actual writing ... */
-    block_fs_fwrite__(block_fs, filename, file_node, ptr, data_size);
-    if (new_node)
-        block_fs_insert_index_node(block_fs, filename, file_node);
+    block_fs_fwrite__(block_fs, filename, block, ptr, data_size);
```

Comment:
It seems that this function does two things. Updates start and end markers, and writes the block. So perhaps we should split it right here:
```cpp
write_start_marker(block_fs);
write_block(block_fs, filename, block, ptr, data_size);
write_end_marker(block_fs);
```

Reply:
Is the start/end marking actually useful? What are the chances that we manage to write the markers successfully but not the data itself? I reckon we remove it entirely.

Reply:
Probably something we can do long-term, but I would probably refactor without changing behavior first.